src/jit/lower.cpp

```diff
@@ -644,7 +644,7 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 GenTreePtr gtCaseBranch = comp->gtNewOperNode(GT_JTRUE, TYP_VOID, gtCaseCond);
                 LIR::Range caseRange    = LIR::SeqTree(comp, gtCaseBranch);
-                currentBBRange->InsertAtEnd(std::move(condRange));
+                currentBBRange->InsertAtEnd(std::move(caseRange));
             }
```

Comment:
wow. looks like we never hit this case because we almost always use jump tables on non-arm platforms. Can you submit this as a separate PR? Or I can. I'd like to see it tested with minSwitchTabJumpCnt=10000 or similar, to force all switches to this path.